lib/std/child_process.zig

```diff
@@ -30,10 +30,20 @@ pub const ChildProcess = struct {
 
     allocator: mem.Allocator,
 
+    /// The write end of the child process's standard input stream.
+    /// Requires `stdin_behavior == StdIo.Pipe`.
```

Comment:
Usage requires `stdin_behavior == StdIo.Pipe` and `stdin` is still null very shortly after process creation.
dito below.

Reply:
It's null after process creation, but if I'm reading this right, it'll be non-null when spawn returns. Instead of documenting a data race, might be better to document that the value is available when spawn returns.

Reply:
> it'll be non-null when spawn returns

Relevant is the parent process, as the child process does not other things in parallel after fork/clone/createProcess.
The parent process sets the file handles around ~20 instructions after execution of the createProcess.
On Unixoids (fork/clone) it might be after a few syscalls and if a parallel thread polls stdout after checking that the subprocess is spawned, it is a race condition.

Reply:
I'm happy to add that if you feel that it's important, but honestly, that seems not super helpful to a user reading the API documentation just trying to figure out how to use this thing.
Please correct me if I'm wrong, but my understanding is that the usage contract is this: set `stdout_behavior`, call `spawn()`, then do stuff with `stdout`.

```
child.stdout_behavior = .Pipe;
child.spawn();
assert(chiild.stdout != null)
```

If a parallel thread is polling `child.stdout` without coordinating with the thread calling `child.spawn()`, there isn't much we can guarantee. (I might even argue that's invalid usage, but I don't know enough about this ecosystem to make that argument.)

If on the other hand, they coordinate the two threads with a synchronization primitive, e.g. the thread calling `spawn()` unlocks a mutex after spawn returns, then they can work with the guarantee we're already giving them in the API documentation: if `stdout_behavior` is `.Pipe`, `child.stdout` will be non-null when `spawn()` returns.

Adding "child.stdout is null for a very short while after spawning the child process" is just going to sow confusion on top of the usage contract ("so can I expect child.stdout to be non-null when spawn returns or not?"), and should perhaps be considered an implementation detail.